src/python/pants/bsp/goal.py

```diff
@@ -26,15 +38,140 @@ class BSPGoal(BuiltinGoal):
     def activated(cls, union_membership: UnionMembership) -> bool:
         return False
 
+    @classmethod
+    def register_options(cls, register):
+        super().register_options(register)
+        register(
+            "--server",
+            type=bool,
+            default=False,
+            advanced=True,
+            help=(
+                "Run the Build Server Protocol server. Pants will receive RPC requests via the console. "
+                "This should only be invoked via the IDE."
+            ),
+        )
+        register(
+            "--runner-env-vars",
+            type=list,
+            member_type=str,
+            default=["PATH"],
+            help=(
+                "Environment variables to set in the BSP runner script when setting up BSP support. "
+                "Entries are either strings in the form `ENV_VAR=value` to set an explicit value; "
+                "or just `ENV_VAR` to copy the value from Pants' own environment.\n\n"
+                "Note: IntelliJ on macOS apparently does not pass through the PATH from the shell, "
+                "and so on macOS at the very least, `PATH` should be explicitly written into the BSP "
+                "runner script via this option."
```

Comment:
Probably worth asking about or opening an issue in the BSP repo about this. It certainly seems like the PATH should be propagated, since in most cases the server will be a standalone process, which would otherwise not have access to environment variables.

Reply:
See https://apple.stackexchange.com/questions/51677/how-to-set-path-for-finder-launched-applications. (The process in question is `launchd`.)